### DISCUSSION TITLE
Questions from quick reference

### DISCUSSION THREAD

**Discussion by reagle** [saulpw/visidata#1845] [Q&A] [answered]
I continue to go over the documentation, especially the clipboard material, for the Anki deck and wonder:

* select rows matching typed value of current cell/row in current column/all visible columns
	* What is the difference between the display value and typed value?
* g+ aggregator    add Ar aggregator No to selected source columns
* gz_ number       adjust widths of all visible columns to Ar number
	* What is "Ar"?
* Memory Sheet (Alt+Shift+M)        browse saved values, including clipboard
	* How does the memory sheet work? Does it keep a history of possible values? I don't think so, it appears to have a row for each type that has been cut or yanked.
* z+ aggregator    display result of aggregator over values in selected rows for current column; store result in Memory Sheet
	* What is the use of this?
* ^Y               paste from cell clipboard
	* what is the cell clipboard?

**Comment by saulpw** [comment, marked as answer]
> What is the difference between the display value and typed value

The typed value is based on the column type, like `int` or `date`, and is used for aggregation, sorting, and computation in general.  The display type is based on the formatter and fmtstr, and is what is shown in the cell, and used for grouping.  (So if you have a date column and want to group by year, you don't have to create a new column with only the year, you can set the fmtstr to `%Y` instead).  The typed value is what's saved in typed formats (like JSON) and the display value is what's saved in text formats (like csv).

> What is "Ar"?

troff detritus.  It should indicate a certain kind of manpage formatting, but apparently here has been misinterpreted as text.  It's a manpage bug.

> What is the use of `z+`?

1.  To do a 'one-off' aggregation over all rows (or all selected rows if any are selected).  Sometimes you just want the total of a column; otherwise you could `+sum` and then `=1` to make a constant column and then freq on that `1` column.  But that's a lot of steps when all you wanted is one value.  So `z+` shows it as a status message.
2. Also, that value is often useful in other computation.  Like you want to use the total as the denominator, perhaps on a different freq sheet.  So the `z+` value is also added to the Memory Sheet, and you can use the name in any expression.

> How does the memory sheet work? Does it keep a history of possible values? I don't think so, it appears to have a row for each type that has been cut or yanked.

It has named values which can be used in any expression.  I don't use it much myself currently, but it could be expanded both in function and documentation.

> what is the cell clipboard?

`zy` and `zx` copy to the internal clipboard.  As they copy only one cell, it goes to the 'cell clipboard'; when you paste only one cell, or Ctrl+Y to paste a value, it uses the most recent value from the cell clipboard.

The clipboard can also have rows on it, but it wouldn't make sense (in my mind) to paste whole rows into a cell or an input prompt.  So they're separate.
